src/_pytest/_io/saferepr.py

```diff
@@ -2,65 +2,59 @@
 import reprlib
 
 
-def _call_and_format_exception(call, x, *args):
+def _format_repr_exception(exc, obj):
+    exc_name = type(exc).__name__
     try:
-        # Try the vanilla repr and make sure that the result is a string
-        return call(x, *args)
-    except Exception as exc:
-        exc_name = type(exc).__name__
-        try:
-            exc_info = str(exc)
-        except Exception:
-            exc_info = "unknown"
-        return '<[{}("{}") raised in repr()] {} object at 0x{:x}>'.format(
-            exc_name, exc_info, x.__class__.__name__, id(x)
-        )
+        exc_info = str(exc)
+    except Exception:
+        exc_info = "unknown"
+    return '<[{}("{}") raised in repr()] {} object at 0x{:x}>'.format(
+        exc_name, exc_info, obj.__class__.__name__, id(obj)
+    )
+
+
+def _ellipsize(s, maxsize):
+    if len(s) > maxsize:
+        i = max(0, (maxsize - 3) // 2)
+        j = max(0, maxsize - 3 - i)
+        return s[:i] + "..." + s[len(s) - j :]
+    return s
 
 
 class SafeRepr(reprlib.Repr):
     """subclass of repr.Repr that limits the resulting size of repr()
     and includes information on exceptions raised during the call.
     """
 
-    def repr(self, x):
-        return self._callhelper(reprlib.Repr.repr, self, x)
-
-    def repr_unicode(self, x, level):
-        # Strictly speaking wrong on narrow builds
-        def repr(u):
-            if "'" not in u:
-                return "'%s'" % u
-            elif '"' not in u:
-                return '"%s"' % u
-            else:
-                return "'%s'" % u.replace("'", r"\'")
+    def __init__(self, maxsize):
+        super().__init__()
+        self.maxstring = maxsize
+        self.maxsize = maxsize
 
-        s = repr(x[: self.maxstring])
-        if len(s) > self.maxstring:
-            i = max(0, (self.maxstring - 3) // 2)
-            j = max(0, self.maxstring - 3 - i)
-            s = repr(x[:i] + x[len(x) - j :])
-            s = s[:i] + "..." + s[len(s) - j :]
-        return s
+    def repr(self, x):
+        try:
+            s = super().repr(x)
+        except Exception as exc:
```

Comment:
This `except` block is [not covered](https://codecov.io/gh/pytest-dev/pytest/compare/499fda234997d65b40e24f6fe3c7341f524b7d97...b47e1679e02029b847a9932e68cdf5db207991b8/diff), can you please add a test for it? Thanks!